lib/kafka/consumer.rb

```diff
@@ -309,6 +321,10 @@ def send_heartbeat_if_necessary
       @heartbeat.send_if_necessary
     end
 
+    def subscribed_partitions
+      @group.subscribed_partitions
+    end
```

Comment:
Remind me again why this is necessary?

Reply:
@dasch because we hold offsets in a buffer for longer than a fetch, which means that the consumer could have been assigned a different partition. We need to check if this consumer instance is still subscribe to that partition and i use this to do this.

```
      when "complete_purchase"
        # We've received all the messages we need, time to save the transactions
        save_transaction(buffer)
        if consumer.subscribed_partitions.include?(message.partition)
          consumer.mark_message_as_processed(message)
          consumer.commit_offsets
        end
        buffer.clear
      end
```

Maybe it's better to this on some kind of callback?

Reply:
Hmm. Nah, I think it's OK.